polymorphic/admin.py

```diff
@@ -136,6 +135,23 @@ def _lazy_setup(self):
         # By not having this in __init__() there is less stress on import dependencies as well,
         # considering an advanced use cases where a plugin system scans for the child models.
         child_models = self.get_child_models()
+        # Check if child_models are provided via new syntax (iterable of models) or compatibility syntax
+        # (iterable of (Model, Admin). When no child_models are provided, assume new syntax.
+        self._compat_mode = len(child_models) and isinstance(child_models[0], (list, tuple))
```

Comment:
there can be a difference between `self.child_models` and `self.get_child_models()` - similar to the way Django views have a `get_form_class()` and `form_class` attribute. It's recommended to always use `get_child_models()` as some packages (e.g. django-fluent-pages) can provide completely dynamic input here.